### PR TITLE
Fix initialization of default date properties

### PR DESCRIPTION
**Pull request by elpoelma**
This PR ensures that the format and value of a data node are correctly initialized to default values if these are not supplied. This PR is a fix to the date issue occuring in https://binnenland.atlassian.net/browse/GN-4183?atlOrigin=eyJpIjoiMzMwNzZlNjgxMDMyNDllM2E0YzIxMzBmY2I3MzU1ZjgiLCJwIjoiaiJ9.

### CODE REVIEW

File: addon/components/rdfa-date-plugin/date-time-picker.ts [hunk 1]
```diff
@@ -7,25 +7,25 @@ import { localCopy } from 'tracked-toolbox';
...
   }
 
   get minutes() {
-    return this.date.getMinutes();
+    return this.date?.getMinutes();
   }
 
   get seconds() {
-    return this.date.getSeconds();
+    return this.date?.getSeconds();
```
Comment (x-m-el) [inline addon/components/rdfa-date-plugin/date-time-picker.ts:28]: This makes sense, but why is it needed if a default is given everywhere (so date shouldn't be undefined anymore)? Future proofing?

**Review by x-m-el — COMMENTED** [review]
I'm guessing the issue would still persist if a default is not given in some new code.
Probably this wouldn't get used outside of the already defined date-node and this isn't a problem though.

File: addon/components/rdfa-date-plugin/date-time-picker.ts (hunk 1 above)
Comment (elpoelma) [reply addon/components/rdfa-date-plugin/date-time-picker.ts:28]: Yes indeed, a default is given everywhere. This is just done for future proofing.

File: addon/plugins/rdfa-date-plugin/nodes/date.ts [hunk 2]
```diff
@@ -16,11 +16,9 @@ const date: (options: DateOptions) => NodeSpec = (options) => {
       mappingResource: {
         default: null,
       },
-      value: {
-        default: new Date().toISOString(),
-      },
+      value: {},
       format: {
-        default: 'dd/MM/yyyy',
+        default: options.formats[0].dateFormat,
```
Comment (abeforgit) [inline addon/plugins/rdfa-date-plugin/nodes/date.ts:21]: Inserting a date in the dummy app is broken, I think it's cause of this line

**Review by abeforgit — CHANGES_REQUESTED** [review]

File: addon/plugins/rdfa-date-plugin/nodes/date.ts (hunk 2 above)
Comment (elpoelma) [reply addon/plugins/rdfa-date-plugin/nodes/date.ts:21]: Should be fixed now.

**Pull request merged**